app/assets/javascripts/discourse/app/components/modal/dismiss-notification-confirmation.js

```diff
@@ -0,0 +1,14 @@
+import Component from "@glimmer/component";
+import { inject as service } from "@ember/service";
+import { action } from "@ember/object";
+
+export default class DismissNotificationConfirmation extends Component {
+  @service modal;
+
+  @action
+  dismiss() {
+    this.args.model.dismissNotifications &&
+      this.args.model?.dismissNotifications();
+    this.modal.close();
```

Comment:
```suggestion
    this.args.closeModal();
```
(and remove service injection)